src/ecmult_impl.h

```diff
@@ -972,12 +972,33 @@ static size_t secp256k1_pippenger_max_points(secp256k1_scratch *scratch) {
     return res;
 }
 
+/* Compute the number and size of batches given the maximum number of points in a batch and the
+ * total number of points */
+static int secp256k1_ecmult_multi_batch_size_helper(size_t *n_batches, size_t *n_batch_points, size_t max_points, size_t n) {
+    if (max_points == 0) {
+        return 0;
+    }
+    if (max_points > ECMULT_MAX_POINTS_PER_BATCH) {
+        max_points = ECMULT_MAX_POINTS_PER_BATCH;
+    }
+    /* Check overflow */
+    if (n > (size_t)-1 - (max_points-1)) {
```

Comment:
Unless we have a reason, I think we'd prefer to use stdint SIZE_MAX here.

Reply:
SIZE_MAX requires C99 as far as I remember.

Reply:
@real-or-random  It is an stdint.h type, like uint32_t.   The codebase uses stdint.h ( https://github.com/bitcoin-core/secp256k1/blob/master/src/util.h#L15)  types everywhere.  In practice every compiler someone would want to use has this header (even non C99 compilers), but if someone did have a compiler without it, they'd have to provide a compatibility header.

Reply:
Okay, I see. It's equivalent but yes, SIZE_MAX is obviously much nicer to read. I was the one who suggested `size_t(-1)` to @jonasnick in the Schnorr signature PR: https://github.com/bitcoin-core/secp256k1/pull/558#pullrequestreview-165077679  Maybe that's why @apoelstra used it in #557 too...